lib/testscript_engine/assertion.rb

```diff
@@ -12,6 +12,13 @@ def initialize(details, outcome)
     end
   end
 
+  class ::Hash
+    def deep_merge(second)
```

Comment:
I don't think we should modify the Hash class for this one use case. I would create a seperate recursive function called `deep_merge` that takes two hashes and merges the first into the second. I really like this approach of merging them to check, so that will preserve that approach. Nice!

Reply:
Agreed. I will separate it as an internal method.